### PR TITLE
Vibankwa/azure.monitor.otel.exporter/fixbreakingtests

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixing some tests which started failing on upgrading otel dependency to latest.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests.AspNetCoreWebApp/ProgramNetCoreAppCompat.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        }
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+    }
+}
+#endif
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests.AspNetCoreWebApp/ProgramNetCoreAppCompat.cs:24]: nit, add an empty line before EOF

**Review by reyang — APPROVED** [review]
LGTM.

**Review by TimothyMothra — APPROVED** [review]
LGTM. review the test run and confirm that the tests actually ran.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/Azure.Monitor.OpenTelemetry.Exporter.Integration.Tests.csproj [hunk 2]
```diff
@@ -18,20 +17,25 @@
...
     To support net461 so we must reference the dependent packages.-->
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" VersionOverride="2.1.1" />
     <PackageReference Include="Microsoft.AspNetCore" VersionOverride="[2.1.1,2.2)" />
     <PackageReference Include="Microsoft.AspNetCore.Hosting" VersionOverride="[2.1.1,2.2)" />
     <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" VersionOverride="[2.1.1,2.2)" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc" VersionOverride="[2.1.1,2.2)" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" VersionOverride="3.0.0" />
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/Azure.Monitor.OpenTelemetry.Exporter.Integration.Tests.csproj:32]: why 3.0.*? we should be using 3.1.*

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Integration.Tests/Azure.Monitor.OpenTelemetry.Exporter.Integration.Tests.csproj]: copy-paste error. fixed it

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**